devito/finite_differences/coefficients.py

```diff
@@ -185,9 +191,23 @@ def generate_subs(i):
             indices, x0 = generate_indices(function, dim, dim.spacing, fd_order,
                                            side=None, stagger=stagger)
 
-            for j in range(len(weights)):
-                subs.update({function._coeff_symbol
-                             (indices[j], deriv_order, function, dim): weights[j]})
+            # NOTE: This implementation currently assumes that indices are ordered
+            # according to their position in the FD stencil. This may not be the
+            # case in all schemes and should be changed such that the weights are
+            # passed as a dictionary of the form {pos: w} (or something similar).
+            if isinstance(weights, np.ndarray):
```

Comment:
Not sure if now or next, but probably it's worth having a wrapper class `Weights` that exposes a common interface and performs the necessary checks. Then:

- you wouldn't need any sort of if-then-else
- you manage to handle the {pos: w} thing easily

I wonder what your thoughts are. I'm fine if we decide to leave this for another day, but perhaps we could add a note somewhere (assuming our thoughts are in sync)

Reply:
Yes, I think a `class weights` sounds like a good option.

If nobody has any strong objections though I'd prefer to leave this for a later date.